source/_ext/dovecot_sphinx.py

```diff
@@ -177,6 +178,8 @@ def resolve_xref(self, env, fromdocname, builder, typ, target, node,
       return make_refnode(builder, fromdocname, todocname, targ, contnode,
                           targ)
     else:
+      logger = logging.getLogger(__name__)
+      logger.warning('Missing cross-reference: %r', target, location=node)
```

Comment:
Is there a `logger.error()` that this could use?

Reply:
logger.warning() is the expected behavior, as Sphinx will emit an event to allow the code to potentially fix/address this.  For our purposes, we want warnings to be treated as errors - but that's run-time configuration.